cli/loader_lib.c

```diff
@@ -161,11 +360,52 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     }
 
     // Introspect to find our own path
-    const char * lib_dir = jl_get_libdir();
+    const char *lib_dir = jl_get_libdir();
 
     // Pre-load libraries that libjulia-internal needs.
     int deps_len = strlen(dep_libs);
-    char * curr_dep = &dep_libs[0];
+    char *curr_dep = &dep_libs[0];
+
+    void *cxx_handle;
+#if defined(_OS_LINUX_)
+    int do_probe = 1;
+    char *probevar = getenv("JULIA_PROBE_LIBSTDCXX");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char *cxxpath = libstdcxxprobe();
+        cxx_handle = dlopen(cxxpath, RTLD_LAZY);
+        char *dlr = dlerror();
+        if (dlr) {
+            fprintf(stderr, "Error, cannot load \"%s\"\ndlopen() - %s\n", cxxpath, dlr);
+            exit(1);
+        }
+        free(cxxpath);
+    }
+    else {
+        char *bundled;
+        size_t len;
+        get_bundled_folder(&bundled, &len);
+        cxx_handle = dlopen(bundled, RTLD_LAZY);
+        free(bundled);
+        char *dlr = dlerror();
+        if (dlr) {
+            fprintf(stderr, "Error, cannot load \"%s\"\ndlopen() - %s\n", bundled, dlr);
+            exit(1);
+        }
+    }   
+#else
+/* void * load_library(const char * rel_path, const char * src_dir, int err) */
+        char *bundled;
+        size_t len;
+        get_bundled_libstdcxx(&bundled, &len);
+        load_library("libstdc++.so", bundled, 1);
+        free(bundled);
+#endif
```

Comment:
Please DRY this, so we aren't doing the same thing 3 times slightly differently. (I posted one code suggestion snippet earlier for how to combine them)